File: Javascript/141/LinkedListCycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    if (head === {}) {
        return false;
    }

    var iterator = head;
    var faster = head;

    while (faster && faster.next) {
        iterator = iterator.next;
        faster = faster.next.next;

        if (!faster) {
            return false;
        }

        if (iterator === faster) {
            return true;
        }
    }

    return false;
};

File: Python/141/LinkedListCycle.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    # @param head, a ListNode
    # @return a boolean
    def hasCycle(self, head):
        iterator = head
        faster = head
        while faster is not None and faster.next is not None:
            iterator = iterator.next
            faster = faster.next.next

            if iterator is faster:
                return True
        return False